### PR TITLE
Refactored view donors page 

### PR DESCRIPTION
**Pull request by Manan-dev**
# Description

Refactored view donors page to use Data table to display data and follows view convention now.

Here is how it looks now:

image
## Relevant issue(s)

#108
#110 
#94 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have assigned reviewers to this PR

### CODE REVIEW

**Comment by Manan-dev** [comment]
Note: I added `@mui/x-data-grid` package from MUI for Data Table component.

**Comment by Manan-dev** [comment]
Note: I added `@mui/x-data-grid` package from MUI for Data Table component.

File: src/views/donorView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    address: donor.address,
+    city: donor.city,
+    state: donor.state,
+    zip: donor.zip,
+    email: donor.email,
+  }));
+
+  return (
+    <div style={{ height: 650, width: '100%' }}>
+      <DataGrid rows={rows} columns={columns} checkboxSelection />
```
Comment (alay10) [inline src/views/donorView/index.tsx:32]: I'm not sure if we are going to want to have the rows be selectable

**Review by alay10 — APPROVED** [review]
I think it looks really good and seems super easy to use. Nice job!

**Review by Trishu-Patel — APPROVED** [review]

**Review by alay10 — APPROVED** [review]
Looks good

**Pull request merged**